### PR TITLE
[BUG] Min Max select query on decimal columns are failing with ClassCastException in Spark beeline

### PR DESCRIPTION
**Pull request by nareshpr**

### CODE REVIEW

File: integration/spark/src/main/scala/org/carbondata/spark/agg/CarbonAggregates.scala [hunk 1]
```diff
@@ -576,7 +614,14 @@ case class MinFunctionCarbon(expr: Expression, base: AggregateExpression1, final
       if (min.isFirstTime) {
         null
       } else {
-        Cast(Literal(min.getValueObject), base.dataType).eval(null)
+        min match {
+          case s: MaxBigDecimalAggregator =>
```
Comment (gvramana) [inline integration/spark/src/main/scala/org/carbondata/spark/agg/CarbonAggregates.scala]: why min matches maxaggreagator

File: integration/spark/src/main/scala/org/carbondata/spark/agg/CarbonAggregates.scala [hunk 2]
```diff
@@ -576,7 +614,14 @@ case class MinFunctionCarbon(expr: Expression, base: AggregateExpression1, final
       if (min.isFirstTime) {
         null
       } else {
-        Cast(Literal(min.getValueObject), base.dataType).eval(null)
+        min match {
+          case s: MinBigDecimalAggregator =>
```
Comment (gvramana) [inline integration/spark/src/main/scala/org/carbondata/spark/agg/CarbonAggregates.scala]: add test cases covering datatypes min,max cases

Comment (nareshpr) [inline integration/spark/src/main/scala/org/carbondata/spark/agg/CarbonAggregates.scala]: Fixed

**Comment by nareshpr** [comment]
retest this please

**Pull request merged**